### ISSUE TITLE
8k/4k support? 😢

### ISSUE THREAD

**Issue by Odinlevi** [sawyerpollard/Minecraft-Weather-New-Tab#9]
Super nice, love at first sight, any thoughts about adding 8k resolution support?

**Comment by sawyerpollard** [comment]
I definitely understand your concern with the resolution of the images. I sourced these images from various websites, and, unfortunately, these were the highest resolution versions that I could find. The hardest part is finding aesthetic photos with high enough resolution. If someone were able to freely share photos for the extension, I would be happy to include them. And, if I have extra time, I can try to load some shaders and take screenshots.

**Comment by Odinlevi** [comment]
Did you tried to ask Reddit for help? I believe there is a major possibility that there will be plenty of nice screenshots

**Comment by GreatSymphonia** [comment]
@Odinlevi Look at this discussion here: #14

**Comment by randy-halim** [comment]
I could possibly provide some photos in 4k/8k with my (semi) decent gaming pc and a screenshot mod I found.
If that’s okay, I’ll see what I can do :p

**Comment by randy-halim** [comment]
After monkeying around for 15 minutes in Forge, I got a 8k screenshot working. I'll open up a pull request sometime soon to integrate this (after correcting the photo and making it so it won't download 100mb files every time...).

In the meantime, here's a sneak peak:
![image](https://user-images.githubusercontent.com/52173775/134450644-c3ad3bfa-1ae0-4ae0-b5bc-d6fc594e27e0.png)

**Comment by GreatSymphonia** [comment]
@randy-halim Wow!  Thanks a lot, don't forget to credit yourself ;)

**Comment by randy-halim** [comment]
> And day and night variations for each, but with the same position and angle.

That sounds very interesting actually! I'll see what I can do

**Comment by randy-halim** [comment]
[Uploaded originals to Google Drive because they're 100mb each ._.](https://drive.google.com/drive/folders/1RZgxrkxyM9wAnETFmxjMIfINu9H55MB8?usp=sharing)

Is this what you're looking for? This was whipped together in 10 minutes, just wanted to make sure I got the right idea.
![plains_raw_night](https://user-images.githubusercontent.com/52173775/135015128-fe79fc6d-b21c-4273-87bd-3b251d63218a.jpg)

**Comment by sawyerpollard** [comment]
> [Uploaded originals to Google Drive because they're 100mb each ._.](https://drive.google.com/drive/folders/1RZgxrkxyM9wAnETFmxjMIfINu9H55MB8?usp=sharing)
> 
> Is this what you're looking for? This was whipped together in 10 minutes, just wanted to make sure I got the right idea.
> ![plains_raw_night](https://user-images.githubusercontent.com/52173775/135015128-fe79fc6d-b21c-4273-87bd-3b251d63218a.jpg)

Wow! I think that looks awesome! It's also really cool having even more times of day besides just day and night.

**Comment by randy-halim** [comment]
> Wow! I think that looks awesome! It's also really cool having even more times of day besides just day and night.

Thanks man! I'll find some more time to do this when I get the chance.
As an aside, since the average 8k image clocks in around 10-15 mb each, should the files be hosted somewhere? Currently the images in the current extension are less than a mb each, which is great. Perhaps using a public GitHub pages from this repo may be the way...

*disclaimer i have no idea how github pages work so take that with a grain of salt.

**Comment by GreatSymphonia** [comment]
@randy-halim hosting them somewhere is definitely an idea but not directly on github, I'll have a VPS in a few weeks that we could use and would be willing to dedicate to the cause if needed

**Comment by sawyerpollard** [comment]
I don't think it would be crazy to host compressed versions of the images on GitHub. I think a couple hundred megabytes of images shouldn't be a problem at this scale. And as for the extension, the images are currently stored locally in a folder. Of course, in the future, if many images are included, they could be downloaded locally to the extension (possibly stored with the Chrome Storage API? I'm not sure of the performance though).

**Comment by GreatSymphonia** [comment]
@sawyerpollard from what I do understand of the Chrome Storage API, it is intended to store parameters or things like that, not whole pictures.

**Comment by GreatSymphonia** [comment]
Taking @randy-halim 's pictures at 10MB each (once compressed), and considering 3 pictures per temperature (8 of them), we would be at 240MB of pictures in total.
If we would tolerate only 4k pictures (8k screens still are pretty rare), I estimate we would be at around 2MB/picture (not exactly because compression) which would bring down the global size of the pictures to 60MB.

This cancels out the issue #15 because having fluid 4k gif would ask for a lot more storage (loop of 3 second at 30 FPS -> 90*2MB*24 = 4GB of gifs compressible to around 2GB)

So, I am for the idea of 4k pictures for everything while still making 8k versions available for the peoples that would like to use those (by using dev mode on chrome and manually adding the extension) and totally dropping the idea of having animated wallpapers.

Fell free to correct my maths or adding new info about the compressed image size, thoseare quick maths done with very few information.

**Comment by sawyerpollard** [comment]
> Taking @randy-halim 's pictures at 10MB each (once compressed), and considering 3 pictures per temperature (8 of them), we would be at 240MB of pictures in total. If we would tolerate only 4k pictures (8k screens still are pretty rare), I estimate we would be at around 2MB/picture (not exactly because compression) which would bring down the global size of the pictures to 60MB.
> 
> This cancels out the issue #15 because having fluid 4k gif would ask for a lot more storage (loop of 3 second at 30 FPS -> 90_2MB_24 = 4GB of gifs compressible to around 2GB)
> 
> So, I am for the idea of 4k pictures for everything while still making 8k versions available for the peoples that would like to use those (by using dev mode on chrome and manually adding the extension) and totally dropping the idea of having animated wallpapers.
> 
> Fell free to correct my maths or adding new info about the compressed image size, thoseare quick maths done with very few information.

Yep, I think that sounds pretty reasonable. I remember when I was testing the extension, I had a lot of problems with cutting down on loading times of images, so ultimately, I settled on 1MB photos (I think progressive JPGs).

I plan on doing some more testing on the maximum image file size that offers fast loading times. I'll also test the WebP image format and see if that offers an improvement.

**Comment by GreatSymphonia** [comment]
Personally, I think we should have 4k pictures regardless of the loading time but this is a personal opinion.

**Comment by randy-halim** [comment]
After playing around at the export settings for Photoshop, we can actually get these files to be chopped down to a reasonable size (1MB) instead of just exporting at "best quality." There are a few jpg artifacts here and there, but they shouldn't be too bad. 

Additionally, having it exported at @1x, @0.5x, and @0.25x means we can have the same images, just for the different screen sizes. @GreatSymphonia you'll get your 4k pictures :p
<details>
<summary>Photoshop's "Very Good" quality</summary>
<br />

at 8k:
![plains_raw_morning@VeryGood_8k](https://user-images.githubusercontent.com/52173775/135152614-e479b801-fcaf-4f79-9723-cf0c3a82e93d.jpg)
at 4k:
![plains_raw_morning-1@VeryGood_4k](https://user-images.githubusercontent.com/52173775/135152636-decfe609-bdfe-4a3d-bc60-30f6c70cb3f6.jpg)
at 2k (1920x1080):
![plains_raw_morning-2@VeryGood_2k](https://user-images.githubusercontent.com/52173775/135152641-bedc7500-729e-4712-8311-21e14fa4de07.jpg)

</details>
<details>
<summary>Photoshop's "Good" quality</summary>
<br />

8k:
![plains_raw_morning@Good_8k](https://user-images.githubusercontent.com/52173775/135153104-f629e34f-9311-4b7c-86eb-ca0c5c3a251b.jpg)
4k:
![plains_raw_morning-1@Good_4k](https://user-images.githubusercontent.com/52173775/135153106-2021482f-9d8e-4ee8-843e-238bd132430f.jpg)
2k:
![plains_raw_morning-2@Good_2k](https://user-images.githubusercontent.com/52173775/135153107-c4c3dd8d-b689-4a8e-9fc7-c5820ed7184f.jpg)

</details>

> This cancels out the issue #15 because having fluid 4k gif would ask for a lot more storage (loop of 3 second at 30 FPS -> 902MB24 = 4GB of gifs compressible to around 2GB)

For #15, I don't think having 4k gifs will work; 1920x1080 sounds more reasonable. However, having a video in the [webm](https://en.wikipedia.org/wiki/WebM) format might work, provided a) the video is relatively short, and b) some tinkering is done to reduce the size to say, 5-10 MB.

**Comment by GreatSymphonia** [comment]
> For #15, I don't think having 4k gifs will work; 1920x1080 sounds more reasonable. However, having a video in the webm format might work, provided a) the video is relatively short, and b) some tinkering is done to reduce the size to say, 5-10 MB.
True, at 1920*1080, it would be doable.  But to capture that and have a good loop, that would be hard.
I would be curious to see such videos
